### PR TITLE
Runner arg parsing regressions

### PR DESCRIPTION
**Pull request by whiteinge**
### What does this PR do?

Fixes regressions caused by #39472.

### What issues does this PR fix or reference?

#40845, #41733.

### Previous Behavior

The initial PR attempted to handle both old- and new-style Runner arg/kwarg handling and clean up redundant parsing logic. I removed the wrong redundant logic. We need to call `load_args_and_kwargs` because it introspects the function and optionally ignores invalid args -- where the regressions came from.

### New Behavior

This replaces `parse_input` with `load_args_and_kwargs` (which calls `parse_input`).

I'm pretty confident this is the right move...but I said that last time.

### Tests written?

Yes.

### CODE REVIEW

**Comment by whiteinge** [comment]
And I just noticed `[WARNING ] /Users/shouse/src/salt/salt/salt/minion.py:301: DeprecationWarning: The list of function args and kwargs should be parsed by salt.utils.args.parse_input() before calling salt.minion.load_args_and_kwargs()`.

Our parsing logic is extremely hard to follow. 😩   Perhaps something we can tackle in a future PR?

**Comment by cachedout** [comment]
Go Go Jenkins!

**Review by rallytime — APPROVED** [review]

**Comment by rallytime** [comment]
@whiteinge and @cachedout I think this looks good and is working well after some smoke testing. I agree that the parsing logic is hard to follow and I have been tripped up by it before. The one thing I am worried about though is that the particular warning that is happening here was removed in 2017.7 (because of the deprecation path). See PR #39646.

So I don't know if this will still work when this is merged forward since it's passing string kwargs potentially. I think a follow up PR will be needed. Maybe @terminalmage has something to add since he did the original deprecation. (I got tripped up on the proper deprecation of this a while ago, and so Erik ended up doing that final removal for 2017.7.)

**Comment by terminalmage** [comment]
Yeah, removing parse_kwargs may cause args to be loaded as data types other than strings. Not sure if this is OK or not.

**Comment by whiteinge** [comment]
Is it correct to say that it is the job of `parse_input` to turn `name=value` into `{'name': 'value'}`? And is it correct to say it is the job of `load_args_and_kwargs` to introspect the function to check if the given params match the function signature?

Why was using the `parse_input` inside `load_args_and_kwargs` put on a deprecation path?
